client/boot/common.js

```diff
@@ -181,6 +190,15 @@ const setRouteMiddleware = () => {
 	} );
 };
 
+const setApplicationMiddleware = () => {
+	page( '*', ( context, next ) => {
+		const app = window.application ? window.application : 'calypso';
```

Comment:
Why are we using `window` here? What's the expected way we're going to be specifying this arg? Should we use something from the framework stack of Calypso - feature/config flags, or some sort of environment variable, or just do it per domain? We already have `CALYPSO_ENV=jetpack-cloud-*`

Reply:
Part of the reason why I am using passing its own variable is because I want to be able to have a way to switch between calypso and cloud just by using a different url and not have to restart calypso in order to test out the things on both calypso and jetpack cloud. 

this makes it much easier to develop/test things across the apps.